src/quacc/atoms/skzcam.py

```diff
@@ -6,35 +6,173 @@
+    adsorbate_vector_from_slab = (
+        adsorbate[0].position - adsorbate_slab[slab_center_idx[0]].position
+    )
+
+    # Add the height of the adsorbate from the slab along the z-direction relative to the first center atom of the slab as defined in the slab_center_idx
+    adsorbate_com_z_disp = (
+        adsorbate_com[2] - adsorbate_slab[slab_center_idx[0]].position[2]
+    )
+    center_position = (
+        np.array([0.0, 0.0, adsorbate_com_z_disp])
+        + slab_center_position
+        - adsorbate_slab[slab_center_idx[0]].position
+    )
+
+    return (
+        adsorbate,
+        slab,
+        slab_first_atom_idx,
+        center_position,
+        adsorbate_vector_from_slab,
+    )
+
+
+@requires(chemshell_module, "ChemShell is not installed")
+def run_chemshell(
+    slab: Atoms,
+    slab_center_idx: int,
+    atom_oxi_states: dict[str, float],
+    chemsh_radius_active: float = 40.0,
+    chemsh_radius_cluster: float = 60.0,
+    chemsh_bq_layer: float = 6.0,
+    write_xyz_file: bool = False,
+):
+    """
+    Run ChemShell to create an embedded cluster from a slab.
+
+    Parameters
+    ----------
+    slab
+        The Atoms object of the slab.
+    slab_center_idx
+        The index of the (first) atom at the center of the slab, this index corresponds to the atom in the slab_center_idx list but adjusted for the slab (which does not contain the adsorbate atoms)
+    atom_oxi_states
+        The oxidation states of the atoms in the slab as a dictionary
+    chemsh_radius_cluster
+        The radius of the total embedded cluster in Angstroms.
+    chemsh_radius_active
+        The radius of the active region in Angstroms. This 'active' region is simply region where the charge fitting is performed to ensure correct Madelung potential; it can be a relatively large value.
+    chemsh_bq_layer
+        The height above the surface to place some additional fitting point charges in Angstroms; simply for better reproduction of the electrostatic potential close to the adsorbate.
+    write_xyz_file
+        Whether to write an XYZ file of the cluster for visualisation.
+    """
+
+    # Translate slab such that first Mg atom is at 0,0,0
+    slab.translate(-slab.get_positions()[slab_center_idx])
+
+    # Convert ASE Atoms to ChemShell Fragment object
+    slab_frag = convert_atoms_to_frag(slab, connect_mode="ionic", dim="2D")
+
+    # Add the atomic charges to the fragment
+    slab_frag.addCharges(atom_oxi_states)
+
+    # Create the chemshell cluster (i.e., add electrostatic fitting charges) from the fragment
+    chemsh_embedded_cluster = slab_frag.construct_cluster(
+        origin=slab_center_idx,
+        radius_cluster=chemsh_radius_cluster / Bohr,
+        radius_active=chemsh_radius_active / Bohr,
+        bq_layer=chemsh_bq_layer / Bohr,
+        adjust_charge="coordination_scaled",
+    )
+
+    # Save the final cluster to a .pun file
+    chemsh_embedded_cluster.save("ChemShell_cluster.pun", "pun")
```

Comment:
You'll want to add an argument to the parent function that specifies where this file is written. We can't have it always write to the current working directory because quacc workflows don't actually change directories into where the calculations are run (because doing `os.chdir` calls break things in parallelism).